brave/brave-ios
# Fix #7523: Select Token to Send view

## Summary of Changes
- Update Send token view to remove the account & network picker, and instead use a single modal for selecting Account to send from, Token to send, and which network.
- The modal contains a network filter, and by default will only show tokens with an available balance.
    - If there are no tokens with an available balance, a `Show Zero Balances` button is displayed that will reveal all user assets for all accounts regardless of balance.
- Designs will change with Wallet v2. Design improvement suggestions for this v1 style are welcome 🙂.

This pull request fixes #7523

## Submitter Checklist:


## Test Plan:

1. Open Send in native Brave Wallet on a Wallet with some token balance(s) on multiple accounts.
2. Tap on the selected token to send from
3. Verify all accounts are displayed, and all of their visible tokens with balances load in as expected. NFTs should be displayed too (assuming you own it / have balance!).
4. Tap on the network filter and select a network to filter by.
5. Verify the displayed accounts / tokens are filtered appropriately.
6. Tap on the network filter and select a network you do not have any token balance on any account.
7. Verify `Show Zero Balances` button is shown in the toolbar. Tap it.
8. Verify zero balance tokens (visible assets) are now displayed for all accounts.
9. Tap on any token in Select Token to Send modal.
10. Verify the selected token, account and network update as expected.
11. Tap on the from token
12. Pick a token on a different account and network. Repeat step 11 as appropriate.
13. Create a send transaction & verify the expected network, account and token is displayed.


## Screenshots:

Send (light) | Send (dark) | Select Token to Send (light) | Select Token to Send (dark)
--|--|--|--
![Send - light mode](https://github.com/brave/brave-ios/assets/5314553/600535f8-5b9d-46fb-97b7-ee900df9bb0c) | ![Send - dark mode](https://github.com/brave/brave-ios/assets/5314553/eae8951a-4f14-4dac-ac79-268800feada2) | ![Select Token to Send - light mode](https://github.com/brave/brave-ios/assets/5314553/75388c1c-4671-44ec-963d-be0ca6970279) | ![Select Token to Send - dark mode](https://github.com/brave/brave-ios/assets/5314553/2c69083a-21c0-435d-94b3-b4dec20b95ff)

Select Token to Send (w/ NFT) | Send (w/ NFT)
--|--
![Select Token to Send - NFT](https://github.com/brave/brave-ios/assets/5314553/d6bfcd9f-e103-42b6-b4b2-6cc8216867b3) | ![Send - NFT](https://github.com/brave/brave-ios/assets/5314553/f1072bf6-4169-4b33-a433-03027db9420b)


https://github.com/brave/brave-ios/assets/5314553/2d853275-d50f-436d-81b2-f79192796c90


## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
